### PR TITLE
Unretire subject by subject ids

### PR DESCRIPTION
**Pull request by yuenmichelle1**
**Changes**
- update unretire subjects by subject set id (updating corresponding operation to grab subject ids from subject set id and then send it over to the worker.)

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/operations/workflows/unretire_subjects.rb [hunk 1]
```diff
@@ -17,7 +21,14 @@ def execute
...
 
     def cached_subject_ids
-      @cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids.push(*subject_set_subject_ids) unless subject_set_subject_ids.empty?
+      @cached_subject_ids = cached_subject_ids
+    end
+
+    def subject_set_subject_ids
+      set_ids ||= Array.wrap(@subject_set_ids) | Array.wrap(@subject_set_id)
```
Comment (camallen) [inline app/operations/workflows/unretire_subjects.rb:30]: no need to memoize here
```suggestion
      set_ids = Array.wrap(@subject_set_ids) | Array.wrap(@subject_set_id)
```

File: app/operations/workflows/unretire_subjects.rb [hunk 2]
```diff
@@ -17,7 +21,14 @@ def execute
     end
 
     def cached_subject_ids
-      @cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids.push(*subject_set_subject_ids) unless subject_set_subject_ids.empty?
+      @cached_subject_ids = cached_subject_ids
```
Comment (camallen) [inline app/operations/workflows/unretire_subjects.rb:26]: I think we can simplify this a little bit - keep the `cached_subject_ids` method but add a layer underneath it to handle the subject_id lookups, e.g. something like
```
  def cached_subject_ids
    # note the union | array operator below will allow us to accept both sets of params :)
    @cached_subject_ids ||= subject_ids | subject_set_subject_ids
  end

  def subject_ids
      Array.wrap(@subject_ids) | Array.wrap(@subject_id)
  end
```

**Review by camallen — COMMENTED** [review]
nice work - some minor comments that might help simplify the code.

**Review by camallen — APPROVED** [review]
LGTM 🎉

**Comment by yuenmichelle1** [comment]
@camallen Thanks! Suggested changes made things look super clean and nice.

**Pull request merged**